qutebrowser/browser/webkit/webkithistory.py

```diff
@@ -48,7 +50,8 @@ def historyContains(self, url_string):
         Return:
             True if the url is in the history, False otherwise.
         """
-        return url_string in self._history.history_dict
+        with debug.log_time('sql', 'historyContains'):
+            return url_string in self._history
```

Comment:
I'm not sure if this is fast enough - maybe it'd make sense to add a cache here, i.e. a simple set which gets populated at runtime, so at least every URL only gets searched in the database once.